core/src/apps/common/passphrase.py

```diff
@@ -0,0 +1,81 @@
+from micropython import const
+
+import storage.device
+from trezor import wire
+from trezor.messages import ButtonRequestType
+from trezor.messages.ButtonAck import ButtonAck
+from trezor.messages.ButtonRequest import ButtonRequest
+from trezor.messages.PassphraseAck import PassphraseAck
+from trezor.messages.PassphraseRequest import PassphraseRequest
+from trezor.ui import ICON_CONFIG, draw_simple
+from trezor.ui.passphrase import CANCELLED, PassphraseKeyboard
+from trezor.ui.text import Text
+
+if __debug__:
+    from apps.debug import input_signal
+
+_MAX_PASSPHRASE_LEN = const(50)
+
+
+def is_enabled() -> bool:
+    return storage.device.is_passphrase_enabled()
+
+
+async def get(ctx: wire.Context) -> str:
+    if is_enabled():
+        return await _request_from_user(ctx)
+    else:
+        return ""
+
+
+async def _request_from_user(ctx: wire.Context) -> str:
```

Comment:
i still think `_request_from_user` and `_get_from_user` should be a single function: first get from user or from host, then check the length

Reply:
Definitely. Fixed in 61e6d3e4843f2a9000956b4e46225868b1da3c3e